### ISSUE TITLE
Does this works? It is not even in npm

### ISSUE THREAD

**Issue by loama** [capacitor-community/auth0#1]
I downloaded it from git and added it into my repo, but it didn't work as expected.
The issue could totally be in my code but I just want to make sure if this is supposed to work yet.

**Comment by christofu** [comment]
Hmmm. Funny this comment arrived just now since I've just been investigating the same thing.
The android and webview parts seem to work great. I just tried to build for ios and discovered that the ios stuff just seems to be started, not even stubbed out fully let alone complete.
Happy to be wrong, but yes, I'd also like to know. I'm poking around to see if I can get the ios parts complete to be working.

**Comment by benji24290** [comment]
Hey guys, i had similar problem. At the end i managed to do it with the browser plugin. It works for me on Android and iOS, but the code is really messy. I also deleted some code that was relevant to redirect login for the webapp.
```javascript
import { Plugins } from "@capacitor/core";
const { Browser, App } = Plugins;
import cryptoRandomString from "crypto-random-string";
import CryptoJS from "crypto-js";
import CONFIG from "../auth/auth0Config";
import { Redirect, useHistory } from "react-router-dom";

function base64URLEncode(str) {
  return str
    .toString("base64")
    .replace(/\+/g, "-")
    .replace(/\//g, "_")
    .replace(/=/g, "");
}

const code_verifier = base64URLEncode(
  cryptoRandomString({ length: 64, type: "base64" })
);
const hash = CryptoJS.SHA256(code_verifier);
const hashString = hash.toString(CryptoJS.enc.Base64);
const code_challenge = base64URLEncode(hashString);

function SignIn() {
  const history = useHistory();
  const [usr, setUsr] = useState(null);
  const [inProgress, setInProgress] = useState(false);

  const loginNative = async () => {
    try {
      //IF USER IS CACHED ON DEVICE
      const userAuthinfo = await getUserAuthInfo()
      setUsr(userAuthinfo);
    } catch (error) {
      //IF USER IS NOT CACHED ON DEVICE
      App.addListener("appUrlOpen", async ({ url }) => {
        App.removeAllListeners();
        if (url) {
          if (!isPlatform("android")) {
            await Browser.close();
          }
          const { search } = new URL(url); 
          const urlParams = new URLSearchParams(search);
          const reqUrl = `https://${CONFIG.auth0Domain}/oauth/token`;
          const response = await fetch(reqUrl, {
            method: "POST",
            headers: {
              Accept: "application/json",
              "Content-Type": "application/json",
            },
            body: JSON.stringify({
              client_id: CONFIG.clientIdMobile,
              code: urlParams.get("code"),
              grant_type: "authorization_code",
              redirect_uri: `${CONFIG.pkgId}://${CONFIG.auth0Domain}/cordova/${CONFIG.pkgId}/callback`,
              code_verifier: code_verifier,
            }),
          });
          let data = await response.json();
          console.log("AUTH_RESPONSE:", data);
          if (data && data.access_token) {
            const userInfo = await getUserInfo(data.access_token);
            console.log(userInfo);
            if (userInfo) {
              setUsr(userInfo);
              try {
                await setUserAuthInfo(userInfo)
              } catch (error) {
                console.log('Failed to cache userAuthInfo')
              }
              history.push("/", { user: getRelevantUserInfo(userInfo) });
            }
          }
          setInProgress(false);
        }
      });
      await Browser.open({
        url: `https://${CONFIG.auth0Domain}/login?state=mystate&client=${CONFIG.clientIdMobile}&protocol=oauth2&scope=openid%20profile%20offline_access&code_challenge_method=S256&response_type=code&redirect_uri=${CONFIG.pkgId}%3A%2F%2F${CONFIG.auth0Domain}%2Fcordova%2F${CONFIG.pkgId}%2Fcallback&code_challenge=${code_challenge}`,
      });
      Browser.addListener("browserFinished", console.log("BROWSER FINISHED"));
      Browser.addListener(
        "browserPageLoaded",
        console.log("BROWSER browserPageLoaded")
      );
    }
  };
 
  const login = async () => {
    setInProgress(true);
    if (isPlatform("hybrid")) {
      loginNative();
    } else {
      await loginWithRedirect();
      setInProgress(false);
    }
  };

 if (usr) {
    console.log("AZURE_AUTH_SUCCESS:", usr);
    console.log("history:",history)
    return (
      <Redirect
        to={{
          pathname: "/",
          state: { user: getRelevantUserInfo(usr) },
        }}
      />
    );
  } else if (!inProgress) {
    return <LoadingScreen />;
  } else {
    return <LoadingScreen />;
  }
}```

**Comment by loama** [comment]
@benji24290 thanks for the snippet, I almost made it work for me, but got stuck in what seemed to be the last step.
Ended up using the password grant flow, here is the docs https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow in case anyone is interested.

**Comment by RavirajWalke** [comment]
@christofu 
Were you able to make it work with ios and How did you copy resources for android and ios builds(As this plugin won't be found with **npx cap sync** ).

**Comment by christofu** [comment]
> Were you able to make it work with ios and How did you copy resources for android and ios builds(As this plugin won't be found with **npx cap sync** ).

I spent a day or so adding the ios stuff, interestingly there is absolutely zero reason why this has to be a capacitor plugin: all the functionality that is implemented in platform-specific code can be easily implemented directly in typescript or js, which is what I did with ios. There were a few Browser related issues (ios kept leaving a blank Browser open after the check was finished) and I probably could have got it working, but I ended up jumping ship to capacitor-oauth2 (and still using auth0) rather than keep plugging away at it.

In short, this incomplete plugin is probably unnecessary and I wouldn't suggest you spent much time here. If nothing else the code snippet quoted above from @benji24290 was helpful if you're not interested in using capacitor-oauth2

**Comment by jcesarmobile** [comment]
I'm archiving the repository since it's been stale for almost 4 years, the maintainer disappeared and nobody has tried to continue the work on the iOS part.
